pandas/core/tools/datetimes.py

```diff
@@ -227,7 +227,11 @@ def _maybe_cache(
         unique_dates = unique(arg)
         if len(unique_dates) < len(arg):
             cache_dates = convert_listlike(unique_dates, format)
-            cache_array = Series(cache_dates, index=unique_dates)
+            # GH#45319
+            try:
+                cache_array = Series(cache_dates, index=unique_dates)
+            except OutOfBoundsDatetime:
+                pass
```

Comment:
Does it make sense just to `return cache_array` here since the `cache_array` cannot be used due to the out of bounds date?